src/components/ebay-bar-chart/component.js

```diff
@@ -0,0 +1,346 @@
+import Highcharts from 'highcharts';
+import accessibility from 'highcharts/modules/accessibility';
+import patternFill from 'highcharts/modules/pattern-fill';
+
+import {
+    chartFontFamily,
+    backgroundColor,
+    gridColor,
+    labelsColor,
+    legendColor,
+    legendInactiveColor,
+    legendHoverColor,
+    tooltipBackgroundColor,
+    tooltipShadows,
+    setSeriesColors,
+} from '../../common/charts/shared';
+
+import { ebayLegend } from '../../common/charts/legend';
+import { eBayColumns } from '../../common/charts/bar-chart';
+
+export default class {
+    onMount() {
+        this._initializeHighchartsExtensions();
+        this._setupCharts();
+    }
+    getContainerId() {
+        return `ebay-bar-chart-${this.id}`;
+    }
+    _initializeHighchartsExtensions() {
+        // enable highcharts accessibility with wrapper function
+        accessibility(Highcharts);
+        // patternFill highcharts wrapper function enables rendering patterns instead of just solid colors
+        patternFill(Highcharts);
+        // add custom legend wrapper function
+        ebayLegend(Highcharts);
+        // add custom columns wrapper to enable rounded bar corners, and stacks with spaces between each stacked point
+        eBayColumns(Highcharts);
+    }
+    _setupCharts() {
+        // check if a single series was passed in for series and if so add it to a new array
+        const series = Array.isArray(this.input.series) ? this.input.series : [this.input.series];
+        // controls rounded corders and spacing at the bottom of data points
+        if (this.input.stacked) {
+            series[0].bottom = true; // set a variable on the first series so it renders rounder corners on the bottom of the bar
+            series[series.length - 1].top = true; // set a variable on the last series to render rounded corner on the top of the bar
+
+            series.forEach((s) => {
+                // used to help link each series to the previous one for stacked views
+                // refer to https://api.highcharts.com/highcharts/series.column.linkedTo
+                s.group = ':previous';
+            });
+        } else {
+            // if not stacked, set the top and bottom flag on each series so the single bar has rounded top and bottom corners
+            series.forEach((s) => {
+                s.top = true;
+                s.bottom = true;
+            });
+        }
+        const config = {
+            title: {
+                text: this.input.title, // set the title that will render above the chart
```

Comment:
Can we set all the input's at the top of the function as variables

Like:
`const title = this.input.title`

And then we can use it later.
This way we can see at the very beginning what `inputs` are used and don't have to dig into the large config.